### PR TITLE
Fix zone reload for new zones

### PR DESCRIPTION
**Pull request by StephenKing**
`pdns_control reload` does not add new zones, as the manpage states.
Therefore, use `pdns_control bind-add-zone $zone` instead.

If a zone is updated, use `bind-reload-now` as well.

### CODE REVIEW

File: providers/zone.rb [hunk 1]
```diff
@@ -23,11 +23,34 @@ def whyrun_supported?
...
+
+  @current_resource.name(new_resource.name)
+  @current_resource.exists = ::File.file?("/etc/powerdns/zones/#{new_resource.name}.zone")
+  @current_resource
+end
+
+# Returns the pdns_control reload command, depending on if
+# we create a new zone or update an existing one.
+def reload_command
+
```
Comment (StephenKing) [inline providers/zone.rb]: Hey @andristeiner, I'm especially interested in your feedback regarding this part. Is this correct? It's hart for me to test zone updates during a single Chef run.

**Comment by StephenKing** [comment]
I pushed the two commits that could have directly gone to _develop_ in the mean time. A pity that Github doesn't update the PR's changes..

File: providers/zone.rb [hunk 2]
```diff
@@ -23,11 +23,34 @@ def whyrun_supported?
...
+  if @current_resource.exists
+    log "Zone #{@new_resource.name} already exists. Doing reload."
+    "pdns_control bind-reload-now #{@new_resource.name}"
+  else
+    log "New zone #{@new_resource.name}. Adding."
+    "pdns_control bind-add-zone #{@new_resource.name}"
```
Comment (StephenKing) [inline providers/zone.rb]: We need to add the path to the zone file in addition...

**Comment by StephenKing** [comment]
closed in favor of #3

File: providers/zone.rb (hunk 1 above)
Comment (andristeiner) [inline providers/zone.rb]: I fire a `/usr/bin/pdns_control reload` after zone changes, however a do not add new zones as most of my zones are coming from a mysql backend.

**Pull request closed without merging**